src/pages/index.js

```diff
@@ -0,0 +1,346 @@
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  
+  if (data.isLiked) { 
+    cardLikedBtn.classList.add("card__like-button_liked");
+  }
+
+  cardLikedBtn.addEventListener("click", (evt) => 
+    handleLike(evt, data._id, cardLikedBtn));
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalCaptionEl.textContent = data.name;
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+  });
+
+  cardDeleteBtn.addEventListener("click", () => 
+    handleDeleteCard(cardElement, data._id, cardDeleteBtn));
+
+  return cardElement;
+}
+
+ 
+
+function handleDeleteCard(cardElement, cardId, cardDeleteBtn) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+  
+  setButtonText(cardDeleteBtn, true, "Delete", "Deleting...");
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  if (avatarDeletetBtn) {
+    setButtonText(avatarDeletetBtn, true, "Delete", "Deleting...");
+  }
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      
+      closeModal(deleteModal);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      setButtonText(avatarDeletetBtn, false, "Delete", "Deleting...");
+    });
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  //change text content to "Saving...""
+  const cardSubmitBtn = evt.submitter; 
+  // cardSubmitBtn.textContent = "Saving...";
+  setButtonText(cardSubmitBtn, true);
+  console.log(setButtonText);
+
+  const updatedUserInfo = {
+    name: editModalNameInput.value,
+    about: editModalDescriptionInput.value,
+  };
+
+  api
+    .editUserInfo(updatedUserInfo) 
+    .then((data) => {
+      profileName.textContent = data.name; 
+      profileDescription.textContent = data.about; 
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      //Call setButtonText instead
+      setButtonText(cardSubmitBtn, false);
+    });
+}
+
+//TODO - implement loading text for all other form submissions
+
+function handleAddCardSumbmit(evt) {
+  evt.preventDefault();
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  const cardEl = getCardElement(inputValues);
+  cardForm.reset();
+  closeModal(cardModal);
+  
+  api.createCard(inputValues) 
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      setButtonText(cardSubmitBtn, true);
+      closeModal(cardModal);  
+    })
+    .catch((err) => {
+      console.error(err);  
+      setButtonText(cardSubmitBtn, false);
+    });
+}
+
+function handleAvatarSubmit(evt){
+  evt.preventDefault();
+  console.log(avatarInput.value);
+  api
+  .editAvatarInfo(avatarInput.value)
+  .then((data) =>{
+    console.log(data.avatar)
+    const avatarImage = document.querySelector(".profile__avatar");
```

Comment:
`    const avatarImage = document.querySelector(".profile__avatar");
`

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine